streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsProducerTest.java

```diff
@@ -74,24 +77,29 @@
     private final MockProducer<byte[], byte[]> nonEosMockProducer = new MockProducer<>(
         cluster, true, new DefaultPartitioner(), byteArraySerializer, byteArraySerializer);
     private final StreamsProducer nonEosStreamsProducer =
-        new StreamsProducer(nonEosMockProducer, false, logContext);
+        new StreamsProducer(nonEosMockProducer, AT_LEAST_ONCE, logContext);
 
-    private final MockProducer<byte[], byte[]> eosMockProducer = new MockProducer<>(
+    private final MockProducer<byte[], byte[]> eosAlphaMockProducer = new MockProducer<>(
```

Comment:
Can you elaborate? I don't understand how this would simplify the test? For example:
```
// existing code
    @Test
    public void shouldInitTxOnEos() {
        assertThat(eosAlphaMockProducer.transactionInitialized(), is(true));
    }

// new code
    @Test
    public void shouldInitTxOnEos() {
        assertThat(eosAlphaStreamsProducer.kafkaProducer().transactionInitialized(), is(true));
    }
```

Why is the new code simpler? In fact, it looks more complex to me?

Reply:
It's simplified because we have a clear mapping from eosAlphaStreamsProducer to its corresponding internal producer, which in the existing code reader always has to build this mapping manually by checking the initialization code.

Reply:
Seem rather subjective. I me it's obvious that one needs to read the test setup code to understand the test.